### PR TITLE
Convert the content plugins to service providers

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts the rest of the content plugins to service providers.

### Testing Instructions
- Enable all content plugins
- Load a module in an article
- Add a page break to an article
- Save the article
- Check if the article is indexed
- Check if the module is loaded on the front end

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@laoneo PHPCS fails due to method names with underscores. It either needs to fix that in the file itself, or you add the files to this rule at the right place: https://github.com/joomla/joomla-cms/blob/4.4-dev/ruleset.xml#L150

**Comment by sandewt** [comment]
> @sandewt do you have time to test this one?

I'll make some time. Testing is phased.

Vote plugin: tested successfully (1)
- works + code review

File: plugins/content/joomla/src/Extension/Joomla.php [hunk 1]
```diff
@@ -136,25 +124,25 @@ public function onContentAfterSave($context, $article, $isNew): void
...
         $default_language = ComponentHelper::getParams('com_languages')->get('administrator');
-        $debug            = $this->app->get('debug_lang');
+        $debug            = $this->getApplication()->get('debug_lang');
 
         foreach ($users as $user_id) {
             if ($user_id != $user->id) {
                 // Load language for messaging
-                $receiver = User::getInstance($user_id);
+                $receiver = $this->getUserFactory()->loadUserById($user_id);
                 $lang     = Language::getInstance($receiver->getParam('admin_language', $default_language), $debug);
```
Comment (heelc29) [inline plugins/content/joomla/src/Extension/Joomla.php:138]: Use `LanguageFactoryInterface` here?

Comment (laoneo) [reply plugins/content/joomla/src/Extension/Joomla.php:138]: In another pr, when we cleanup Language::getInstance. Thanks for the hint!

Comment (heelc29) [reply plugins/content/joomla/src/Extension/Joomla.php:138]: ok

**Comment by richard67** [comment]
@MacJoom Hmm, not really helpful to do a branch update without restoring @heelc29 's test result in the issue tracker using the "Alter test" button. A branch update always resets the human tests counter, and so we won't find this PR later after another human test when we are searching for PR's which have 2 tests for setting them RTC. I have restored the test in the issue tracker now, but you can't always count on me being around.

**Comment by sandewt** [comment]
It's great that this PR has been merged, but I can't keep up with the pace of testing. See above. :sweat:

**Comment by laoneo** [comment]
No problem, target was to ship it with the first alpha. Hopefully we get then a bigger audience which will check it.

**Pull request merged**